src/fourcipp/utils/dict_utils.py

```diff
@@ -126,3 +127,211 @@ def compare_nested_dicts_or_lists(
         )
 
     return True
+
+
+def _get_dict(nested_dict, keys, optional=True):
+    """Return dict entry within a nested dict by keys.
+
+    In case a list is encountered, this function yields over every entry.
+
+    Args:
+        nested_dict (dict, list): Dict to iterate. Due to recursiveness, this can also be a list
+        keys (list): List of keys to access
+        optional (bool): If the entry is part of a collection that does no exist as it is optional
+
+    Yields:
```

Comment:
Maybe use returns here to be consistent with other files
```suggestion
    Returns:
```

Reply:
I used the yields to indicate that this is a generator. This means you have to iterate through it, for example using 
```python
for entry in _get_dict(**arguments):
     print(entry)
```

This functionality seems strange, but keep in mind if you have a 4C metadata type list, then you might have to change every object of that list. Think about `RESULT DESCRIPTION`, e.g. you want to change a parameter within each entry.

Reply:
Thanks for the explanation 🙏🏻